media-proxy/src/rdma_session.c

```diff
@@ -345,9 +340,19 @@ static void *rx_rdma_frame_thread(void *arg)
     libfabric_ctx *rdma_ctx = s_ctx->rdma_ctx;
     ep_ctx_t *cp_ctx = s_ctx->ep_ctx;
 
-    printf("%s(%d), start\n", __func__, s_ctx->idx);
+    while (!atomic_load_explicit(&s_ctx->shm_ready, memory_order_acquire) && !s_ctx->stop) {
+        DEBUG("%s memif not ready\n", __func__);
+        usleep(1000);
+    }
+
+    printf("%s(%d), RX RDMA thread started\n", __func__, s_ctx->idx);
     while (!s_ctx->stop) {
+
         ep_recv_buf(cp_ctx, cp_ctx->data_buf, s_ctx->transfer_size);
+        if (!atomic_load_explicit(&s_ctx->shm_ready, memory_order_acquire)) {
+            DEBUG("memif disconnected");
```

Comment:
I suggest to avoid printing that constantly in the loop.
```suggestion
```

Reply:
ok, I will remove it entirely as a log is printed in place where the variable is changed.